app/utils/SwapCategoriesUtils.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import config.ApplicationConfig
+
+object SwapCategoriesUtils {
+
+  private def swapCategories[A](configList: List[String], sortedSpendDataMap: Map[String, A]): List[(String, A)] =
+    configList.map { category =>
+      {
+        val spendData = sortedSpendDataMap(category)
+        (category, spendData)
+      }
+    }
+
+  def getAndSwapCategories[A](
+    appConfig: ApplicationConfig,
+    taxYear: Int,
+    sortedSpendDataList: List[(String, A)]): List[(String, A)] = {
+    val orderOfSpendCategories = try {
```

Comment:
Based on previous comments you can probably remove most of this error handling code (try/catch) and just use the empty list as None